server/src/main/java/de/uftos/services/StudentService.java

```diff
@@ -174,7 +188,10 @@ public void deleteStudents(String[] ids) {
     }
 
     studentGroupRepository.saveAll(studentGroups);
-    this.repository.deleteAllById(Arrays.stream(ids).toList());
 
+    new ConstraintInstanceDeleter(constraintSignatureRepository, constraintInstanceRepository)
+        .removeAllInstancesWithArgumentValue(ids);
+
+    this.repository.deleteAllById(Arrays.stream(ids).toList());
```

Comment:
Same issue with `delete()` as in GradeService.

Reply:
Same response as in GradeService.